pkg/controller/utils/utils.go

```diff
@@ -122,3 +122,16 @@ func LogLevel(err error) log.Level {
 		err = cause
 	}
 }
+
+// AddLabel sets the specified label on the object. If the label key is already present, then it is over written with the value. If the labels map is nil, a new labels map is created.
+func AddLabel(object metav1.Object, key, value string) {
+	objLabels := object.GetLabels()
+	if objLabels == nil {
+		object.SetLabels(map[string]string{key: value})
```

Comment:
Sorry, my suggestion of using `labels` directly when passed into `SetLabels` does not work as well here now that the function is `AddLabel`.

I suggest the following.
```
if objLabels == nil {
  objLabels = map[string]string}{}
}
```